File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
    	
    	ListNode previous_node = null;
    	ListNode current_node = head;
    	ListNode next_node = null;
    	
    	while(current_node != null) {
    		
    		next_node = current_node.next;
    		current_node.next = previous_node;
    		previous_node = current_node;
    		current_node = next_node;
    		
    	}
    	return previous_node;
    }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        previous_node = None
        current_node = head
        next_node = None

        while current_node:

            next_node = current_node.next

            current_node.next = previous_node #reverse

            previous_node = current_node
            current_node = next_node
                        
        return previous_node